docker_mcp/tools/containers.py

```
@@ -199,6 +224,112 @@ async def list_containers(
+        try:
+            client = await self.context_manager.get_client(host_id)
+            if client is None:
+                return {
+                    "success": False,
+                    "error": f"Could not connect to Docker on host {host_id}",
+                }
+
+            # Step 1: Try exact match by ID/name (fast, uses Docker API directly)
+            try:
+                container = await asyncio.to_thread(client.containers.get, container_identifier)
+                return {"success": True, "container": container}
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)